File: backend/src/main/java/ru/chsergeig/shoppy/dao/AccountOrderRepository.java
package ru.chsergeig.shoppy.dao;

import org.jooq.Configuration;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Repository;
import ru.chsergeig.shoppy.jooq.tables.daos.AccountsOrdersDao;
import ru.chsergeig.shoppy.jooq.tables.pojos.Accounts;

import java.util.List;

import static ru.chsergeig.shoppy.jooq.Tables.ACCOUNTS;
import static ru.chsergeig.shoppy.jooq.Tables.ACCOUNTS_ORDERS;

@Repository
public class AccountOrderRepository extends AccountsOrdersDao {

    public AccountOrderRepository(@Autowired Configuration configuration) {
        super(configuration);
    }

    public List<Accounts> getAccountsByOrderId(Integer orderId) {
        return ctx().select(ACCOUNTS.asterisk())
                .from(ACCOUNTS)
                .leftJoin(ACCOUNTS_ORDERS).on(ACCOUNTS_ORDERS.ACCOUNT_ID.eq(ACCOUNTS.ID))
                .where(ACCOUNTS_ORDERS.ORDER_ID.eq(orderId))
                .fetchInto(Accounts.class);
    }

}

File: backend/src/main/kotlin/ru/chsergeig/shoppy/dao/AccountOrderRepository.kt
package ru.chsergeig.shoppy.dao

import org.jooq.DAO
import org.jooq.Record2
import ru.chsergeig.shoppy.jooq.tables.pojos.Accounts
import ru.chsergeig.shoppy.jooq.tables.pojos.AccountsOrders
import ru.chsergeig.shoppy.jooq.tables.records.AccountsOrdersRecord

interface AccountOrderRepository :
    DAO<AccountsOrdersRecord?, AccountsOrders?, Record2<Int?, Int?>> {

    fun getAccountsByOrderId(
        orderId: Int?
    ): List<Accounts>
}
